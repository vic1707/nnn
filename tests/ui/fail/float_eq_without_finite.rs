#![expect(unused_imports, reason = "Not what we're testing.")]
#![expect(unexpected_cfgs, reason = "Dunno how to generate them properly.")]
use nnn::nnn;

#[nnn(derive(PartialEq, Eq))]
struct Float(f64);

fn main() {}

// Invalid because `finite` validator is gated behind a `cfg`
// While `derive(Eq)` isn't.
#[nnn(derive(PartialEq, Eq), cfg(test, validators(finite)))]
struct FiniteFloat(f64);

// Invalid because `finite` validator is gated behind a `cfg`
// While `derive(Eq)` is behind a different one.
#[nnn(
    derive(PartialEq),
    cfg(toto, validators(finite)),
    cfg(test, derive(Eq))
)]
struct FiniteFloat2(f64);
