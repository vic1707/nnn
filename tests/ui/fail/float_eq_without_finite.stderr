error[E0277]: the trait bound `f64: std::cmp::Eq` is not satisfied
 --> tests/ui/fail/float_eq_without_finite.rs:6:14
  |
5 | #[nnn(derive(PartialEq, Eq))]
  |                         -- in this derive macro expansion
6 | struct Float(f64);
  |              ^^^ the trait `std::cmp::Eq` is not implemented for `f64`
  |
  = help: the following other types implement trait `std::cmp::Eq`:
            i128
            i16
            i32
            i64
            i8
            isize
            u128
            u16
          and $N others
note: required by a bound in `AssertParamIsEq`
 --> $RUST/core/src/cmp.rs
  |
  | pub struct AssertParamIsEq<T: Eq + ?Sized> {
  |                               ^^ required by this bound in `AssertParamIsEq`
  = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `f64: std::cmp::Eq` is not satisfied
  --> tests/ui/fail/float_eq_without_finite.rs:13:20
   |
12 | #[nnn(derive(PartialEq, Eq), cfg(test, validators(finite)))]
   |                         -- in this derive macro expansion
13 | struct FiniteFloat(f64);
   |                    ^^^ the trait `std::cmp::Eq` is not implemented for `f64`
   |
   = help: the following other types implement trait `std::cmp::Eq`:
             i128
             i16
             i32
             i64
             i8
             isize
             u128
             u16
           and $N others
note: required by a bound in `AssertParamIsEq`
  --> $RUST/core/src/cmp.rs
   |
   | pub struct AssertParamIsEq<T: Eq + ?Sized> {
   |                               ^^ required by this bound in `AssertParamIsEq`
   = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)
